src/systems/collision.rs

```diff
@@ -222,37 +240,56 @@ impl<'s> System<'s> for PincerCollisionSystem {
     );
 
     fn run(&mut self, data: Self::SystemData) {
-        let (entities, mut pincers, mut colliders, mut collidees, mut motions,
-            mut directions, transforms, prefab_list, lazy_update, ctx) = data;
+        let (
+            entities,
+            mut pincers,
+            mut colliders,
+            mut collidees,
+            mut motions,
+            mut directions,
+            transforms,
+            prefab_list,
+            lazy_update,
+            ctx,
+        ) = data;
 
-        for (entity, _, motion, collider, collidee, direction, transform) in
-            (&entities, &mut pincers, &mut motions, &mut colliders,
-                &mut collidees, &mut directions, &transforms).join() {
-            if collider.has_collided {
+        for (entity, _, motion, collider, collidee, direction, transform) in (
+            &entities,
+            &mut pincers,
+            &mut motions,
+            &mut colliders,
+            &mut collidees,
+            &mut directions,
+            &transforms,
+        )
+            .join()
+        {
+            let change_direction = match &collider.collision {
+                CollisionSource::Boundary => true,
```

Comment:
I would prefer to have the direction change logic within the match block. Any reason you prefer this way?